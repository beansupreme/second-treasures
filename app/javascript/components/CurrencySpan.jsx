import React from "react"

function CurrencySpan(props) {
  let passedInValue = props.number;
  let displayNumber = "";

  let number = parseFloat(passedInValue);
  if (isNaN(number)) {
    displayNumber = '';
  } else {
    displayNumber = number.toFixed(2);
  }    

  return (
    <span>${displayNumber}</span>
  )
}

export default CurrencySpan;
