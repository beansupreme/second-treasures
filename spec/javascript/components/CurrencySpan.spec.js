import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow} from 'enzyme';
import CurrencySpan from 'components/CurrencySpan'

describe('<CurrencySpan />', () => {

  it('renders a string as expected', () => {
    const tree = renderer.create(<CurrencySpan number={"10.00"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a number as expected', () => {
    const tree = renderer.create(<CurrencySpan number={10.00} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can format a string into currency', () => {
    const wrapper = mount(<CurrencySpan number={"10.55"} />);
    expect(wrapper).toHaveHTML('<span>$10.55</span>')
  });

  it('can format a string with one decimal into currency', () => {
    const wrapper = mount(<CurrencySpan number={"10.0"} />);
    expect(wrapper).toHaveHTML('<span>$10.00</span>')
  });

  it('can format a number into currency', () => {
    const wrapper = mount(<CurrencySpan number={10.00} />);
    expect(wrapper).toHaveHTML('<span>$10.00</span>')
  });

  it('can handle an empty string', () => {
    const wrapper = mount(<CurrencySpan number={''} />);
    expect(wrapper).toHaveHTML('<span>$</span>')
  });
});