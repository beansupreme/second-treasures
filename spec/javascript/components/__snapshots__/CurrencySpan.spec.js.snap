// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencySpan /> renders a number as expected 1`] = `
<span>
  $
  10.00
</span>
`;

exports[`<CurrencySpan /> renders a string as expected 1`] = `
<span>
  $
  10.00
</span>
`;
